models/fivetran_platform__audit_table.sql

```diff
@@ -46,128 +46,141 @@ with base as (
+            when event_subtype = 'records_modified' then {{ fivetran_log.fivetran_log_json_parse(string='message_data', string_path=['operation_type']) }} 
+            else cast(null as {{ dbt.type_string() }})
+        end as operation_type,
 
-add_connection_info as (
+        cast(case 
+            when event_subtype = 'records_modified' then {{ fivetran_log.fivetran_log_json_parse(string='message_data', string_path=['count']) }} 
+            else null
+        end as {{ dbt.type_bigint() }}) as row_count
 
-    select 
-        sync_log.*,
-        connection.connection_name,
-        connection.destination_id,
-        connection.destination_name
-    from sync_log 
-    left join connection
-        on connection.connection_id = sync_log.connection_id
+    from base
 ),
 
 sync_timestamps as (
 
     select
         connection_id,
-        connection_name,
         table_name,
         event_subtype,
-        destination_id,
-        destination_name,
-        created_at as write_to_table_start,
-        min(case when event_subtype = 'write_to_table_end' then created_at else null end) 
-            over (partition by connection_id, table_name order by created_at ROWS between CURRENT ROW AND UNBOUNDED FOLLOWING) as write_to_table_end,
+        created_at,
+        schema_name,
+        operation_type,
+        row_count,
+
+        case 
+            when event_subtype = 'write_to_table_start' then 
+            min(case when event_subtype = 'write_to_table_end' then created_at else null end) 
+                over (partition by connection_id, table_name order by created_at ROWS between CURRENT ROW AND UNBOUNDED FOLLOWING) 
+        else null end as write_to_table_end,
 
+        case 
+            when event_subtype = 'write_to_table_start' then 
         max(case when event_subtype = 'sync_start' then created_at else null end) 
-            over (partition by connection_id order by created_at ROWS between UNBOUNDED PRECEDING and CURRENT ROW) as sync_start,
+            over (partition by connection_id order by created_at ROWS between UNBOUNDED PRECEDING and CURRENT ROW) 
+        else null end as sync_start,
 
+        case 
+            when event_subtype = 'write_to_table_start' then 
+        min(case when event_subtype = 'sync_start' then created_at else null end) 
+            over (partition by connection_id order by created_at ROWS between CURRENT ROW AND UNBOUNDED FOLLOWING)
+        else null end as next_sync_start,
+
+        case 
+            when event_subtype = 'write_to_table_start' then 
         min(case when event_subtype = 'sync_end' then created_at else null end) 
-            over (partition by connection_id order by created_at ROWS between CURRENT ROW AND UNBOUNDED FOLLOWING) as sync_end, -- coalesce with next_sync_start
+            over (partition by connection_id order by created_at ROWS between CURRENT ROW AND UNBOUNDED FOLLOWING) 
+        else null end as sync_end -- coalesce with next_sync_start
```

Comment:
Not important to address before pre-release, but let's standardize the casing of this logic. Code should be in all lowercase.